Add `largeFileCallback` support

Juice10 opened the pull request:
Adds `largeFileCallback` support which allows you to do something with the large files if you'd like.

jscheid commented:
Hey @Juice10, I'm not affiliated with this project but wanted to say thanks, this is exactly what I need (for copying large remote assets to a local directory). Just wanted to ask why your docs have this note in them?

> Make sure you have `strict` set to `false` when using this.

I'd like to keep `strict` set to `true` because I want to be alerted about broken URLs and such. I've tried using your branch with `strict=true` and all seems to be fine. What am I missing?

Juice10 commented:
Hey @jscheid my absolute pleasure! 
Because of how I implemented it, it will trigger an error whenever it comes across a large file. The strict note was added because it could be a little confusing otherwise.

Juice10 commented:
@jscheid thanks to your feedback I rewrote it a little so it doesn't trigger "Too Big" errors when you set a largeFileCallback and I removed the `strict` comment from the docs.

jscheid commented on lib/image.js:
```diff
@@ -60,8 +62,8 @@ async function getDataUri(file, options) {
-async function resolve(filepath, {assetPaths, maxFileSize, resolver = () => undefined}) {
-  const filter = (resource) => assertSize(resource, maxFileSize);
+async function resolve(filepath, {assetPaths, maxFileSize, largeFileCallback, resolver = () => undefined}) {
+  const filter = (resource) => assertSize(resource, maxFileSize, !largeFileCallback);
```
This could also be:

```js
  const filter = (resource) => largeFileCallback ? resource : assertSize(resource)
```

Then you don't have to touch the other function.

Juice10 replied:
Thanks for the tip. Unfortunately will give you a callback on all files, not just the large ones.
